library/core/src/str/mod.rs

```diff
@@ -1160,6 +1160,12 @@ impl str {
     /// The [pattern] can be a `&str`, [`char`], a slice of [`char`]s, or a
     /// function or closure that determines if a character matches.
     ///
+    /// Note that there is a footgun to this method when using a slice of [`char`]s.
+    /// Some users may expect that a slice of chars will behave similarly to a `&str` with this method.
+    /// That is not currently the case. When you pass a slice of [`char`]s to this method, it will return true
+    /// if any of the [`char`]s in the slice is the first [`char`] of this string slice. It does not work for
+    /// sequentially comparing a slice of [`char`]s to a string slice. See the second example below.
```

Comment:
I don't think this wording (with "footgun", "some users", "currently", etc.) is very useful.

How about instead just adding a bit more detail to the paragraph above? Maybe something in this direction?

```diff
-    /// The [pattern] can be a `&str`, [`char`], a slice of [`char`]s, or a
-    /// function or closure that determines if a character matches.
+    /// The [pattern] can be a `&str` to check for a prefix,
+    /// or a [`char`], slice of [`char`]s, or function or closure 
+    /// to check the first character for a match.
```